dz4/MainClass.java

```diff
@@ -0,0 +1,152 @@
+
+    public static void printMap() {
+        System.out.println("  1 2 3");
+        for (int i = 0; i < map.length; i++) {
+            System.out.print((i + 1) + " ");
+            for (int j = 0; j < map[i].length; j++) {
+                System.out.print(map[i][j] + " ");
+            }
+            System.out.println();
+        }
+    }
+
+    public static boolean isCellEmpty(int x, int y) {
+        if (x < 0 || x > 2 || y < 0 || y > 2) return false;
+        if (map[x][y] == '*') return true;
+        return false;
+    }
+
+    public static boolean isMapFull() {
+        for (int i = 0; i < map.length; i++) {
+            for (int j = 0; j < map[i].length; j++) {
+                if (map[i][j] == '*') return false;
+            }
+        }
+        return true;
+    }
+
+    public static boolean checkWin(char symb) {
+
+        /*
+         if (map[0][0] == symb && map[0][1] == symb && map[0][2] == symb) return true;
+         if (map[1][0] == symb && map[1][1] == symb && map[1][2] == symb) return true;
+         if (map[2][0] == symb && map[2][1] == symb && map[2][2] == symb) return true;
+         if (map[0][0] == symb && map[1][0] == symb && map[2][0] == symb) return true;
+         if (map[0][1] == symb && map[1][1] == symb && map[2][1] == symb) return true;
+         if (map[0][2] == symb && map[1][2] == symb && map[2][2] == symb) return true;
+         if (map[0][0] == symb && map[1][1] == symb && map[2][2] == symb) return true;
+         if (map[0][2] == symb && map[1][1] == symb && map[2][0] == symb) return true;
+         return false;
+         */
+        int checkCell;
+        for (int i = 0; i < map.length; i++) {      // Цикл проверяет строки на наличие победы
+            checkCell = 0;
+            for (int j = 0; j < map[i].length; j++) {
+                if (map[i][j] == symb) checkCell++;
+            }
+            if (checkCell == map.length) return true;
+        }
+
+        for (int i = 0; i < map.length; i++) {      // Цикл проверяет столбцы на наличие победы
+            checkCell = 0;
+            for (int j = 0; j < map[i].length; j++) {
+                if (map[j][i] == symb) checkCell++;
+            }
+            if (checkCell == map.length) return true;
+        }
+
+        checkCell = 0;                               // Цикл проверяет диагональ сверху вниз на наличие победы
+        for (int i = 0; i < map.length; i++) {
+            for (int j = 0; j < map[i].length; j++) {
+                if ((i == j) && (map[i][j] == symb)) checkCell++;
+            }
+            if (checkCell == map.length) return true;
+        }
+
+        checkCell = 0;                                // Цикл проверяет диагональ снизу вверх на наличие победы
+        for (int i = 0; i < map.length; i++) {
+            for (int j = 0; j < map[i].length; j++) {
+                if ((i + j == map.length - 1) && (map[i][j] == symb)) checkCell++;
+            }
+            if (checkCell == map.length) return true;
+        }
```

Comment:
У тебя квадратный массив, можно проще.
А можно еще оптимизировать, чтобы циклы не крутились целиком всё время. Надо не на "==" проверять, а наоборот "!=". Можешь подумать как сделать. 
```suggestion
        checkCell = 0;
        for (int i = 0; i < map.length; i++) {
            if (map[i][i] == symb) checkCell++;
        }
        if (checkCell == map.length) {
            return true;
        }

        checkCell = 0;
        for (int i = 0; i < map.length; i++) {
            if (map[map.length - 1 - i][i] == symb) checkCell++;
        }
        if (checkCell == map.length) {
            return true;
        }
```